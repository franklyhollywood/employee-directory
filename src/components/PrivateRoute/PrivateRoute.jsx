export default function PrivateRoutes() {
  return <div></div>;
}
