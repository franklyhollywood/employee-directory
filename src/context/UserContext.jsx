import { useContext, createContext, useState } from 'react';

const userContext = createContext();

const userProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <userContext.Provider value={{ user, setUser }}>
      {children}
    </userContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(userContext);

  if (context === undefined) {
    throw new Error('Incorrect Placement');
  }
  return context;
};

export { userProvider, useUser };
