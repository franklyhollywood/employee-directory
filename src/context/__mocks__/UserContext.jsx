import { useContext, createContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(mockUser ? { email: mockUser.email } : {});

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const UserCtx = useContext(UserContext);

  if (UserCtx === undefined) {
    throw new Error('Must be wrapped in a userprovider to use useUser');
  }
  return UserCtx;
};

export { UserProvider, useUser };
