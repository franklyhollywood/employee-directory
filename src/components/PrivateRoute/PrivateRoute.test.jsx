import { render } from '@testing-library/react';
import PrivateRoute from './PrivateRoute.jsx';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext.jsx';

jest.mock('../../services/users.js');
jest.mock('../../context/UserContext.jsx');

it('should render private route', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <PrivateRoute location="/" />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
