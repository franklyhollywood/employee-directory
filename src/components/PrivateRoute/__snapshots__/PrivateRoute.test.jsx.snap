// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render private route 1`] = `<div />`;
